### PR TITLE
Update MetaIO

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
@dzenanz The Modules/ThirdParty/MetaIO/UpdateFromUpstream.sh script produced conflict for me. I just used resolved the conflict with "theirs" strategy and committed. I'm unsure what problem you had with the script?

**Review by aylward — APPROVED** [review]

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
The update script produced thousands of conflicts. Right-clicking and choosing "resolve using HEAD" took 20 minutes and produced thousands of deleted files. Maybe related to version of git in use? Or default options?

File: Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h [hunk 1]
```diff
@@ -117,9 +117,9 @@ class METAIO_EXPORT MetaTube : public MetaObject
 
     void PrintInfo(void) const override;
 
-    void CopyInfo(const MetaObject * _object) override;
+    void CopyInfo(const MetaTube * _object);
 
-    void Clear(void) override;
+    void Clear(void);
```
Comment (N-Dekker) [inline Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h:122]: The `override` specifier that is removed here does seem useful to me!

**Comment by N-Dekker** [comment]
I'm very glad to to that my MetaIO fix is picked up so quickly, thanks @blowekamp, @dzenanz , @aylward

File: Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h (hunk 1 above)
Comment (hjmjohnson) [reply Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h:122]: Agreed.  This probably needs to be added again in the upstream version.

Approving to not let perfection be the enemy of the good.

**Review by hjmjohnson — APPROVED** [review]

**Comment by dzenanz** [comment]
MetaIO does not see a lot of changes, and most of them are driven by the needs of ITK. So ITK pretty closely follows latest MetaIO.

**Comment by blowekamp** [comment]
> The update script produced thousands of conflicts. Right-clicking and choosing "resolve using HEAD" took 20 minutes and produced thousands of deleted files. Maybe related to version of git in use? Or default options?

This is some info about the platform I used:
 git version 2.17.1
 Ubuntu 18.04.5 LTS

**Comment by dzenanz** [comment]
I have git version 2.26.0.windows.1 on up-to-date Win10.

**Comment by blowekamp** [comment]
> I have git version 2.26.0.windows.1 on up-to-date Win10.

As I recall the update from upstream scripts do not work on OS X either.

**Comment by dzenanz** [comment]
It works for most repos. I think git got confused with MetaIO because ITK's MetaIO reader has some tests where were created by copying them from MetaIO repo and then updating. Since those tests get deleted upon integration of MetaIO in ITK, git matches those tests as copy+modify operation confusing its other heuristics.

File: Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h (hunk 1 above)
Comment (hjmjohnson) [reply Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h:122]: I think the removal of this override will cause compiler warnings.

Comment (dzenanz) [reply Modules/ThirdParty/MetaIO/src/MetaIO/src/metaTube.h:122]: I think @aylward removed it. Maybe he should pitch in?

**Comment by hjmjohnson** [comment]
FYI: https://github.com/Kitware/MetaIO/pull/96 fixes the warning on macOS.

**Comment by hjmjohnson** [comment]
Mac OS failure not related:  961 - itkJPEG2000Test06 (Failed)

**Pull request merged**